money.go

```diff
@@ -27,12 +27,34 @@ func defaultUnmarshalJSON(m *Money, b []byte) error {
 	if err != nil {
 		return err
 	}
-	ref := New(int64(data["amount"].(float64)), data["currency"].(string))
+
+	amountRaw := data["amount"]
+	var amount float64
+	if amountRaw != nil {
+		amount = amountRaw.(float64)
+	}
```

Comment:
to assert if map has a key you could use this syntax
```go
	if amountRaw, ok := data["amount"]; ok {
		amount = amountRaw.(float64)
	}
```

same with currency below

Reply:
Ok cool, I've updated this.